Maths/Determinant.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Given a square matrix, find its determinant.
+ *
+ * For more info: https://en.wikipedia.org/wiki/Determinant
+ *
+ * @param {number[[]]} matrix - Two dimensional array of integers.
+ * @returns {number} - An integer equal to the determinant.
+ *
+ * @example
+ * const squareMatrix = [
+ *                          [2,3,4,6],
+ *                          [5,8,9,0],
+ *                          [7,4,3,9],
+ *                          [4,0,2,1]
+ *                      ];
+ *
+ * const result = determinant(squareMatrix);
+ * // The function should return 858 as the resultant determinant.
+ */
+
+const subMatrix = (matrix, i, j) => {
+  let matrixSize = matrix[0].length
+  if (matrixSize === 1) {
+    return matrix[0][0]
+  }
+  let subMatrix = []
+  for (let x = 0; x < matrixSize; x++) {
```

Comment:
I'd swap `x` (column) and `y` (row), assuming you want to view matrices as being in row-major order.

Reply:
Can you please explain a bit more as I quite don't understand the swaping, as currently all the submatrices are row-major.

Reply:
@appgurueu this is non-blocking I presume?

Reply:
> @appgurueu this is non-blocking I presume?

Yeah, this is non-blocking. It doesn't affect correctness, it's just that I find these loop variables to be named slightly confusingly. I think they should be named the other way around: Outer loop should be over y (row), inner loop should be over x (column), giving `matrix[y][x]` for indexing: You're accessing the y-th row (y as row index) and x-th column (x as column index). Otherwise you have x as row index and y as column index, which I find confusing.